AutoUsing/Program.cs

```diff
@@ -6,54 +6,79 @@
 using System.Reflection;
 using System.Runtime.InteropServices;
 using System.Text;
+
 using Microsoft.VisualBasic;
+
 using Newtonsoft.Json;
 
 namespace AutoUsing
 {
     class Program
     {
-        private static Type[] TypesFrom(string path)
-        {
-            return Assembly.LoadFrom(path).GetTypes();
-        }
+        static StdProxy Proxy { get; set; }
+        static AssemblyScanner Scanner { get; set; }
 
-        private static List<Type> GetAllTypes()
+        static void Main(string[] args)
         {
-            var types = typeof(int).Assembly.GetExportedTypes().ToList();
-            types.AddRange(typeof(Console).Assembly.GetExportedTypes());
-            types.AddRange(typeof(File).Assembly.GetExportedTypes());
-            return types;
-        }
+            Proxy = new StdProxy();
 
-        const string csdata = @"C:\Users\natan\Desktop\Auto-Using\src\csdata\";
+            Proxy.EditorDataReceived += (s, e) =>
+            {
+                /*
+                    {"Command":"ping","Arguments":""}
+                    {"Command":"scan","Arguments":"/Volumes/Workspace/csharp-extensions/Auto-Using/AutoUsing/bin/Debug/netcoreapp2.1/AutoUsing.dll"}
+                    {"Command":"getAllTypes","Arguments":""}
+                    ...
+                */
+                Request req = e.Data;
 
-        const string csreferences = csdata + "csReferences.ts";
-        const string extensionMethods = csdata + "csExtensionMethods.ts";
-        const string hierachies = csdata + "csHierachies.ts";
+                switch (req.Command)
+                {
+                    case "scan":
+                        Scanner = new AssemblyScanner();
 
-        
+                        if (Scanner.LoadAssembly(req.Arguments) is false)
+                            Proxy.WriteData(Errors.CannotLoadAssembly, false);
 
-        static void Main(string[] args)
-        {
-            var assemblyScanner = new AssemblyScanner();
-            var referencesJson = JsonConvert.SerializeObject(assemblyScanner.GetAllReferences(), Formatting.Indented);
-            var extensionsJson = JsonConvert.SerializeObject(assemblyScanner.GetAllExtensionMethods(), Formatting.Indented);
-            var hierachiesJson = JsonConvert.SerializeObject(assemblyScanner.GetAllHierarchies(), Formatting.Indented);
-
-
-            File.WriteAllText(csreferences, $"export const _CSHARP_REFERENCES = {referencesJson};");
-            File.WriteAllText(extensionMethods, $"export const _CSHARP_EXTENSION_METHODS : ExtendedClass[] = {extensionsJson};");
-            File.WriteAllText(hierachies, $"export const _CSHARP_CLASS_HIEARCHIES : ClassHiearchies[] = {hierachiesJson};");
-            // y.ass
+                        Proxy.WriteData();
```

Comment:
There should probably be an empty response object so the argument is not left blank